biggus/__init__.py

```diff
@@ -839,10 +891,114 @@ def _getitem_full_keys(self, keys):
             sub_array = self.array
         else:
             sub_array = self.array[tuple(array_keys)]
-        return BroadcastArray(sub_array, new_broadcast_dict)
+        return BroadcastArray(sub_array, new_broadcast_dict, leading_shape)
+
+    @classmethod
+    def broadcast_arrays(cls, array1, array2):
+        """
+        Given two arrays, if broadcasting needs to take place on either arrays
```

Comment:
Does this sentence make sense to you @rhattersley? Would it be better changing ``, if`` for ``and``?